### ISSUE TITLE
BibKnowledge: import or link to textual KB file

### ISSUE THREAD

**Issue by jirikuncar** [inveniosoftware-attic/invenio-knowledge#18]
@jeromecaffaro commented on [Wed Apr 30 2014](https://github.com/inveniosoftware/invenio/issues/1215)

Originally on 2012-11-21

It would be useful for BibKnowledge to use knowledge base files as input source, such as BibConvert KB files (one rule per line, triple-dashes separated from-to values). Upon creation of such KB, only the path to the KB file would be recorded, and mappings would be resolved using this file. A caching mechanism could be implemented to make sure that a value is always returned, even if the input source is not available (eg. network issue, AFS, file deleted, etc.). It would not be possible to update such KB via the BibKnowledge interface (read-only).

In addition BibKnowledge should also be able to import such input file (one-short import, instead of linking to the file) to create a regular KB (editable, but not synchronized with the KB file).

See also related #1214

---

@jalavik commented on [Fri Sep 19 2014](https://github.com/inveniosoftware/invenio/issues/1215#issuecomment-56159851)
### Proposal:

Port the knowledge-base functionality to `next` using the concept of load/export (via mixer) and configurable storage backends (e.g. redis, SQLAlchemy). Keeping also a registry of kbs on file + dynamic knowledge-bases.

Then we can add a manager in `inveniomanage` to reload the knowledge-bases (`inveniomanage kbs reload`) when needed and it would be automatically triggered upon `database create`.

Then it can be discussed if the web interface should also be ported to allow for viewing and editing of kbs. 

_In short: this feature request would be handled by this port_

@jirikuncar @jmartinm 

---

@hachreak commented on [Fri Jul 24 2015](https://github.com/inveniosoftware/invenio/issues/1215#issuecomment-124449501)

can we move the issue to https://github.com/inveniosoftware/invenio-knowledge ? :)
